pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -543,21 +287,193 @@ func IngressConfig(client client.Client, controller containerjfr.ReconcileContai
 	result, err = controller.Reconcile(req)
-	IngressConfig(client, controller, req, minimal)
+	ingressConfig(client, controller, req, minimal)
 
 	result, err = controller.Reconcile(req)
 	Expect(err).ToNot(HaveOccurred())
 	Expect(result).To(Equal(reconcile.Result{}))
 }
+
+func checkOwner(object metav1.Object) {
+	ownerReferences := object.GetOwnerReferences()
+	Expect(ownerReferences[0].Kind).To(Equal("ContainerJFR"))
+	Expect(ownerReferences[0].Name).To(Equal("containerjfr"))
+}
+
+func expectCertificates(client client.Client, controller containerjfr.ReconcileContainerJFR) {
+	req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+	result, err := controller.Reconcile(req)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(result).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+	certNames := []string{"containerjfr", "containerjfr-ca", "containerjfr-grafana"}
+	for _, certName := range certNames {
+		cert := &certv1.Certificate{}
+		err := client.Get(context.Background(), types.NamespacedName{Name: certName, Namespace: "default"}, cert)
+		Expect(err).ToNot(HaveOccurred())
+	}
+}
+
+func expectRoutes(client client.Client, controller containerjfr.ReconcileContainerJFR, minimal bool) {
+	req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+	result, err := controller.Reconcile(req)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(result).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+
+	// Update certificate status
+	makeCertificatesReady(client)
+	initializeSecrets(client)
+
+	// Check for routes, ingress configuration needs to be added as each
+	// one is created so that they all reconcile successfully
+	result, err = controller.Reconcile(req)
+	ingressConfig(client, controller, req, minimal)
+}
+
+func expectPVC(client client.Client, controller containerjfr.ReconcileContainerJFR, minimal bool) {
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, pvc)
+	Expect(err).To(HaveOccurred())
+
+	reconcileFully(client, controller, minimal)
+
+	err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, pvc)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Compare to desired spec
+	expectedPvc := resource_definitions.NewPersistentVolumeClaimForCR(test.NewContainerJFR())
+	Expect(pvc.ObjectMeta.Name).To(Equal(expectedPvc.ObjectMeta.Name))
+	Expect(pvc.ObjectMeta.Namespace).To(Equal(expectedPvc.ObjectMeta.Namespace))
+	Expect(pvc.Spec.AccessModes).To(Equal(expectedPvc.Spec.AccessModes))
+	Expect(pvc.Spec.StorageClassName).To(Equal(expectedPvc.Spec.StorageClassName))
+
+	pvcStorage := pvc.Spec.Resources.Requests["storage"]
+	expectedPvcStorage := expectedPvc.Spec.Resources.Requests["storage"]
+	Expect(pvcStorage.Equal(expectedPvcStorage)).To(BeTrue())
+
+	checkOwner(&pvc.ObjectMeta)
```

Comment:
Does just passing `pvc` work here? Same for the other resource types.

Also I was thinking you could do like a `checkMetadata` (or similar), and check not only owner references, but Name, Namespace, Labels, Annotations, to further reduce duplication.

Reply:
If I use metav1.Object then I have to pass in the metadata itself.

Reply:
I refactored the objectMeta comparisons into checkMetadata.